virtual_machines.go

```diff
@@ -152,23 +151,27 @@ func (v *VirtualMachine) Migrate(target, targetstorage string) (task *Task, err
 	return NewTask(upid, v.client), nil
 }
 
-func (v *VirtualMachine) Clone(name, target string) (newid int, task *Task, err error) {
+func (v *VirtualMachine) Clone(params *VirtualMachineCloneOptions) (newid int, task *Task, err error) {
 	var upid UPID
-	cluster, err := v.client.Cluster()
-	if err != nil {
-		return newid, nil, err
+
+	if params == nil {
+		params = &VirtualMachineCloneOptions{}
 	}
 
-	newid, err = cluster.NextID()
-	if err != nil {
-		return newid, nil, err
+	if params.NewID == 0 {
```

Comment:
oh ya i like this much more with the `0` check, I do want this to make some repetitive stuff in the proxmox api easier and letting you set this yourself is fine but pass 0 and we gen one is a great touch.